packages/dart/npt_flutter/lib/features/profile/bloc/profile_bloc.dart

```diff
@@ -256,12 +256,108 @@ class ProfileBloc extends LoggingBloc<ProfileEvent, ProfileState> {
         uuid,
       );
     } finally {
-      await npt?.done;
-      cancel?.call();
-      App.navState.currentContext?.read<ProfilesRunningCubit>().invalidate(
-        uuid,
-      );
+      if (npt != null) {
+        await npt.done;
+        cancel?.call();
+        App.navState.currentContext?.read<ProfilesRunningCubit>().invalidate(
+          uuid,
+        );
+ 
+        // If keep-alive is enabled and the session ended (but the profile wasn't explicitly stopped),
+        // retry the connection after a delay
+        if (profile.keepAlive && state is ProfileStarted) {
+          await _retryConnection(emit, profile, atClient, atSign, settings);
+        } else {
+          emit(ProfileLoaded(uuid, profile: profile));
+        }
+      }
+    }
+  }
+
+  Future<void> _retryConnection(
+    Emitter<ProfileState> emit,
+    Profile profile,
+    AtClient atClient,
+    String atSign,
+    Settings settings,
+  ) async {
+    // Wait 5 seconds before retrying, similar to npt binary
+    await Future.delayed(const Duration(seconds: 5));
```

Comment:
Why do we wait 5 seconds in npt binary? Is this something we even need to do. A short delay is fine, but 5 seconds seems excessive.

Reply:
cc: @gkc